dask_expr/io/_delayed.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+import functools
+from collections.abc import Iterable
+from typing import TYPE_CHECKING, Literal
+
+from dask.dataframe.dispatch import make_meta
+from dask.dataframe.utils import check_meta
+from dask.delayed import Delayed, delayed
+
+from dask_expr import new_collection
+from dask_expr._expr import Expr, PartitionsFiltered
+from dask_expr.io import BlockwiseIO
+
+if TYPE_CHECKING:
+    import distributed
+
+
+class _DelayedExpr(Expr):
+    # Wraps a Delayed object to make it an Expr for now. This is hacky and we should
+    # integrate this properly...
+    # TODO
+
+    def __init__(self, obj):
+        self.obj = obj
+        self.operands = [obj]
+
+    @property
+    def _name(self):
+        return self.obj.key
+
+    def _layer(self) -> dict:
+        return self.obj.dask.to_dict()
+
+
+class FromDelayed(PartitionsFiltered, BlockwiseIO):
+    _parameters = ["meta", "user_divisions", "verify_meta", "_partitions"]
+    _defaults = {
+        "meta": None,
+        "_partitions": None,
+        "user_divisions": None,
+        "verify_meta": True,
+    }
+
+    def dependencies(self):
+        return self.dfs
+
+    @functools.cached_property
+    def dfs(self):
+        return self.operands[len(self._parameters) :]
+
+    @functools.cached_property
+    def _meta(self):
+        if self.operand("meta") is not None:
+            return self.operand("meta")
+
+        return delayed(make_meta)(self.dfs[0]).compute()
+
+    def _divisions(self):
+        if self.operand("user_divisions") is not None:
+            return self.operand("user_divisions")
+        else:
+            return (None,) * (len(self.dfs) + 1)
+
+    def _filtered_task(self, index: int):
+        key = self.dfs[index]._name
+        if self.verify_meta:
+            return (
+                functools.partial(check_meta, meta=self._meta, funcname="from_delayed"),
+                key,
+            )
+        else:
+            return identifiy, key
```

Comment:
Sorry, you gotta change this as well.

```suggestion
            return identity, key
```